plugins/operators/__init__.py

```diff
@@ -1416,6 +1416,202 @@ def execute(self, ctx):
         ctx.dataset.create_index(field_name, unique=unique, wait=False)
 
 
+class ConvertIndexToUnique(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="convert_index_to_unique",  # Must match what's in fiftyone.yml
+            label="Convert an existing index to unique",
+            description="Convert an existing non-unique index to a unique index in the dataset if possible or identify any unique key violations that would occur.",
+            allow_immediate_execution=True,
+            allow_delegated_execution=True,
+            default_choice_to_delegated=False,
+        )
+
+    def resolve_input(self, ctx):
+        import re
+
+        inputs = types.Object()
+        indexes = ctx.dataset.get_index_information()
+
+        if not indexes:
+            inputs.str(
+                "index_name",
+                label="This dataset has no non-unique indexes to update",
+                view=types.Warning(),
+            )
+            return types.Property(
+                inputs,
+                invalid=True,
+            )
+
+        indexes = [
+            i
+            for i, v in indexes.items()
+            if not re.match(r"\.?id$", i) and "unique" not in v
+        ]
+
+        index_selector = types.DropdownView()
+        for key in indexes:
+            index_selector.add_choice(key, label=key)
+
+        inputs.str(
+            "index_name",
+            label="Index",
+            description="The existing index to make unique",
+            view=index_selector,
+            required=True,
+        )
+        inputs.bool(
+            "dry_run",
+            required=False,
+            default=False,
+            label="Check for unique key violations only (dry run)",
+        )
+
+        return types.Property(
+            inputs,
+            view=types.View(label="Convert existing index to unique"),
+        )
+
+    def execute(self, ctx):
+        field = ctx.params.get("index_name")
+        dry_run_only = ctx.params.get("dry_run")
+        result = {
+            "field": field,
+            "dry_run": str(bool(dry_run_only)),
+            "status": None,
+            "error": None,
+        }
+        if not field:
+            result["status"] = "error"
+            result["error"] = "No index name provided."
+            return result
+
+        if field not in (
+            all_index_info := ctx.dataset.get_index_information()
+        ):
+            result["status"] = "error"
+            result["error"] = f"Index '{field}' does not exist in the dataset."
+            return result
+
+        index_info = all_index_info[field]
+
+        db = ctx.dataset._sample_collection.database
+        collection_name = (
+            ctx.dataset._frames_collection_name
+            if field.startswith(ctx.dataset._FRAMES_PREFIX)
```

Comment:
It's possible for an image dataset to have a `frames.foo` field, so you need slightly different logic here to *only* switch to frame collection processing if in fact the index refers to a frame-level field.

Here's how that is done in `drop_index()`:
https://github.com/voxel51/fiftyone/blob/21c77ec954b144e24b739e8a3d43c0bcebaf292f/fiftyone/core/collections.py#L10586